lib/plugins/plugin/plugin.js

```diff
@@ -0,0 +1,343 @@
+            let serverlessFileContent = fs.readFileSync(serverlessFilePath).toString();
+
+            // load the service and parse it to JSON
+            const parsedYaml = YAML.load(serverlessFileContent);
+            // remove the plugin from the array if the plugins array is available
+            if (parsedYaml.plugins) {
+              const newPluginsArray = {
+                plugins: [],
+              };
+
+              newPluginsArray.plugins = _.pull(parsedYaml.plugins, pluginName);
+              if (!_.isEmpty(newPluginsArray.plugins)) {
+                // replace the plugins definition in the serverless file
+                serverlessFileContent = serverlessFileContent
+                  .replace(/(plugins:)+(\s)+(- .*\s*)+/, YAML.dump(newPluginsArray));
+              } else {
+                // remove the plugins definition from the serverless file
+                serverlessFileContent = serverlessFileContent
+                  .replace(/(plugins:)+(\s)+(- .*\s*)+/, '');
+              }
+
+              // write the file back to the disc
+              fs.writeFileSync(serverlessFilePath, serverlessFileContent);
+              this.serverless.cli.log(`Successfully uninstalled "${pluginName}"`);
+            }
+          } else {
+            const message = 'An error occurred while uninstalling your plugin. Please try again...';
+            this.serverless.cli.log(message);
+          }
+        } else {
+          const message = `Plugin "${pluginName}" not found. Did you spell it correct?`;
+          this.serverless.cli.log(message);
+        }
+      });
+  }
+
+  list() {
+    return BbPromise.bind(this)
+      .then(this.getPlugins)
+      .then((plugins) => this.display(plugins));
+  }
+
+  search() {
+    return BbPromise.bind(this)
+      .then(this.getPlugins)
+      .then((plugins) => {
+        // filter out plugins which match the query
+        const regex = new RegExp(this.options.query);
+
+        const filteredPlugins = plugins.filter((plugin) =>
+          (plugin.name.match(regex) || plugin.description.match(regex))
+        );
+
+        // print a message with the search result
+        const pluginCount = filteredPlugins.length;
+        const query = this.options.query;
+        const message = `${pluginCount} plugin(s) found for your search query "${query}"\n`;
+        this.serverless.cli.consoleLog(chalk.yellow(message));
+
+        return filteredPlugins;
+      })
+      .then((plugins) => {
+        this.display(plugins);
+      });
+  }
+
+  // helper methods
+  validate() {
+    if (!this.serverless.config.servicePath) {
+      throw new this.serverless.classes
+        .Error('This command can only be run inside a service directory');
+    }
+
+    return BbPromise.resolve();
+  }
+
+  getServerlessFilePath() {
+    const servicePath = this.serverless.config.servicePath;
+    const serverlessYmlFilePath = path.join(servicePath, 'serverless.yml');
```

Comment:
The serverless file can also be a JSON file => serverless.yaml, serverless.yml and serverless.json should be searched here